kyo-core/js/src/main/scala/kyo/KyoAppPlatformSpecific.scala

```diff
@@ -0,0 +1,41 @@
+package kyo
+
+import kyo.Maybe.Absent
+import kyo.Maybe.Present
+
+abstract class KyoAppPlatformSpecific extends KyoApp.Base[Async & Resource & Abort[Throwable]]:
+
+    private var previousFiberIO: Maybe[Fiber[Throwable, Unit] < IO] = Absent
+
+    final override protected def run[A: Flat](v: => A < (Async & Resource & Abort[Throwable]))(using Frame): Unit =
+        import AllowUnsafe.embrace.danger
+        val currentFiberIO =
+            for
+                currentFiber <- handle(v).flatMap(_.mapResult { result =>
+                    IO(Result(if !result.exists(().equals(_)) then println(pprint.apply(result).plainText)))
+                })
+                sleepFiber <- Async.run(Async.sleep(timeout))
+                raceFiber  <- Fiber.race(Seq(currentFiber.get, sleepFiber.get))
+            yield raceFiber
+        previousFiberIO = previousFiberIO match
+            case Absent => Present(currentFiberIO)
+            case Present(fiberIO) =>
+                val concatenatedFiberIO = fiberIO.flatMap(_.flatMap(_ => currentFiberIO))
+                Present(concatenatedFiberIO)
+        if maybeProc.isEmpty then
+            maybeProc = Present(() =>
+                previousFiberIO match
+                    case Present(fiberIO) =>
+                        val combinedFiberIO =
+                            for
+                                keepAliveFiber <- Clock.repeatWithDelay(1.hours)(())
+                                fiber          <- fiberIO
+                                combineFiber   <- Fiber.race(Seq(keepAliveFiber.get, fiber.get))
+                            yield combineFiber
+                        val _ = IO.Unsafe.run(Async.run(combinedFiberIO.get)).eval
```

Comment:
I don't think you need to use `Async.run` or `combinedFiberIO.get` but I could be wrong.

Reply:
most of these back and forth `fiber.get` and `Async.run` are just to convert between `Fiber[E, A]` and `A < (Async & Abort[E])` to use the method. I feel like some APIs can be update here:
- `Fiber.race` should receive `Fiber` seq, currently it's receiving `A < (Async & Abort[E])`
- There is no `timeout` or `sleep` for `Fiber`, only available now on `A < (Async & Abort[E])`